scipy/special/_erfinv.h

```diff
@@ -0,0 +1,62 @@
+#ifndef ERF_INV_H
+#define ERF_INV_H
+
+/*
+ * mconf configures NANS, INFINITYs etc. for cephes and includes some standard
+ * headers. Although erfinv and erfcinv are not defined in cephes, erf and erfc
+ * are. We want to keep the behaviour consistent for the inverse functions and
+ * so need to include mconf.
+ */
+#include "cephes/mconf.h"
+
+/*
+ * Inverse of the error function.
+ *
+ * Computes the inverse of the error function on the restricted domain
+ * -1 < y < 1. This restriction ensures the existence of a unique result
+ * such that erf(erfinv(y)) = y.
+ */
+double erfinv(double y) {
+    if (cephes_isnan(y)) {
+        sf_error("erfinv", SF_ERROR_DOMAIN, NULL);
+        return NPY_NAN;
+    }
+    else if (y <= -1) {
```

Comment:
We want to return NaN for `y < -1` and `-NPY_INFINITY` only when `y == -1`.

Reply:
The current behaviour is to return negative infinity:

```python
>>> import scipy
>>> from scipy.special import erfinv
>>> scipy.__version__, erfinv(-100)
('1.3.1', -inf)
```
 Should I be changing the existing behaviour?

Reply:
Not on master because I fixed it recently:

```
>>> import scipy
>>> scipy.__version__
'1.5.0.dev0+6644a54'
>>> import scipy.special as sc
>>> sc.erfinv(-100)
nan
```

Reply:
:+1: I'll (re)fix it. I will be out for most of today but I should get the changes in tonight.